src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -115,9 +118,11 @@ public override TypeSyntax VisitDynamicType(IDynamicTypeSymbol symbol)
             public override TypeSyntax VisitFunctionPointerType(IFunctionPointerTypeSymbol symbol)
             {
                 // TODO(https://github.com/dotnet/roslyn/issues/39865): generate the calling convention once exposed through the API
-                var parameters = symbol.Signature.Parameters.Select(p => p.Type)
-                    .Concat(SpecializedCollections.SingletonEnumerable(symbol.Signature.ReturnType))
-                    .SelectAsArray(t => SyntaxFactory.Parameter(SyntaxFactory.MissingToken(SyntaxKind.IdentifierToken)).WithType(t.GenerateTypeSyntax()));
+                var parameters = symbol.Signature.Parameters.Select(p => (p.Type, RefKindModifiers: CSharpSyntaxGenerator.GetParameterModifiers(p.RefKind)))
+                    .Concat(SpecializedCollections.SingletonEnumerable((
+                        Type: symbol.Signature.ReturnType,
+                        RefKindModifiers: CSharpSyntaxGenerator.GetParameterModifiers(symbol.Signature.RefKind, forFunctionPointerReturnParameter: true))))
+                    .SelectAsArray(t => SyntaxFactory.Parameter(SyntaxFactory.MissingToken(SyntaxKind.IdentifierToken)).WithModifiers(t.RefKindModifiers).WithType(t.Type.GenerateTypeSyntax()));
```

Comment:
I can't decide if this is a thing of beauty or a a horrific abuse of our language. :smile:

Reply:
Beauty. Definitely beauty.